File: net/lib/rules/BooleanRules.cs
namespace Albert.Lib.Rules;

public class BooleanTrueRule() : IBaseRule<bool>
{
    public string ErrorMessage() => "must be true";

    public bool Validate(bool value) => value == true;
}

public class BooleanFalseRule() : IBaseRule<bool>
{
    public string ErrorMessage() => "must be false";

    public bool Validate(bool value) => value == false;
}

File: js/lib/rules/booleanRules.ts
import { BaseRule } from '.';

export class BooleanTrueRule implements BaseRule<boolean> {
  errorMessage() {
    return 'must be true';
  }

  validate(value?: boolean) {
    return value == true;
  }
}

export class BooleanFalseRule implements BaseRule<boolean> {
  errorMessage() {
    return 'must be false';
  }

  validate(value?: boolean) {
    return value == false;
  }
}
